Source/WebKit/UIProcess/haiku/BackingStoreHaiku.cpp

```diff
@@ -29,34 +33,101 @@
 #if USE(HAIKU)
 
 #include "UpdateInfo.h"
-#include <WebCore/NotImplemented.h>
+
+#include <Rect.h>
+#include <WebCore/IntRect.h>
+#include <WebCore/ShareableBitmap.h>
 
 namespace WebKit {
+using namespace WebCore;
 
-static const Seconds s_scrollHysteresisDuration { 300_ms };
+// BackingStore stores and updates a bitmap of the rendered webpage.
 
 BackingStore::BackingStore(const WebCore::IntSize& size, float deviceScaleFactor)
     : m_size(size)
     , m_deviceScaleFactor(deviceScaleFactor)
-    , m_scrolledHysteresis([this](PAL::HysteresisState state) { if (state == PAL::HysteresisState::Stopped) m_scrollSurface = nullptr; }, s_scrollHysteresisDuration)
+    , m_bitmap(BRect(0, 0, size.width() * deviceScaleFactor, size.height() * deviceScaleFactor), B_RGBA32, true)
+    , m_view(m_bitmap.Bounds(), "BackingStore", 0, 0)
 {
+	m_bitmap.AddChild(&m_view);
 }
 
-BackingStore::~BackingStore() = default;
+BackingStore::~BackingStore()
+{
+    m_bitmap.RemoveChild(&m_view);
+}
 
-void BackingStore::paint(PlatformPaintContextPtr cr, const WebCore::IntRect& rect)
+void BackingStore::paint(BView* into, const WebCore::IntRect& rect)
 {
-    notImplemented();
+    // Paint the contents of our bitmap into the BView.
```

Comment:
could the bitmap be set permanently in the view by using SetViewBitmap? Or would that create other problems? (such as display flickering or displaying unfinished rendering of pages)